File: backend/app/api/schemas/offer.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.domain.enums.offer_state import OfferState
from app.domain.enums.work_mode import WorkMode


class CompanyBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    sector: str | None = None
    main_location: str | None = None


class SourceBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    source_type: str


class OfferOut(BaseModel):
    """Schéma de sortie complet d'une offre normalisée."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    normalized_title: str
    normalized_description: str | None = None
    contract_type: str | None = None
    duration_months: int | None = None
    location_text: str | None = None
    work_mode: WorkMode | None = None
    education_level: str | None = None
    published_at: datetime | None = None
    deadline_at: datetime | None = None
    current_state: OfferState
    is_active: bool
    global_score: float | None = None
    action_score: float | None = None
    score_justification: dict | None = None
    offer_url: str | None = None
    created_at: datetime
    updated_at: datetime

    # Relations
    company: CompanyBrief | None = None
    primary_source: SourceBrief | None = None


class OfferListItem(BaseModel):
    """Schéma allégé pour la liste des offres."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    normalized_title: str
    contract_type: str | None = None
    duration_months: int | None = None
    location_text: str | None = None
    work_mode: WorkMode | None = None
    current_state: OfferState
    global_score: float | None = None
    action_score: float | None = None
    published_at: datetime | None = None
    created_at: datetime

    company: CompanyBrief | None = None
    primary_source: SourceBrief | None = None


class PaginatedOffers(BaseModel):
    """Réponse paginée — format standard V2."""

    items: list[OfferListItem]
    total: int
    page: int
    page_size: int
    has_next: bool

File: frontend/src/types/offer.ts
// Types TypeScript alignés sur les schémas Pydantic du backend

export type OfferState =
  | "DETECTED"
  | "RAW_STORED"
  | "NORMALIZED"
  | "DEDUPLICATED"
  | "ANALYZED"
  | "REJECTED"
  | "QUALIFIED"
  | "DRAFT_REQUESTED"
  | "DRAFT_PREPARED"
  | "READY_FOR_REVIEW"
  | "SUBMISSION_IN_PROGRESS"
  | "SUBMITTED"
  | "CLOSED";

export type WorkMode = "ONSITE" | "HYBRID" | "REMOTE";

export type SortBy = "created_at" | "relevance_score";

export interface CompanyBrief {
  id: string;
  name: string;
  sector: string | null;
  main_location: string | null;
}

export interface SourceBrief {
  id: string;
  name: string;
  source_type: string;
}

export interface OfferListItem {
  id: string;
  normalized_title: string;
  contract_type: string | null;
  duration_months: number | null;
  location_text: string | null;
  work_mode: WorkMode | null;
  current_state: OfferState;
  is_active: boolean;
  global_score: number | null;
  tags: string[] | null;
  published_at: string | null;
  created_at: string;
  company: CompanyBrief | null;
  primary_source: SourceBrief | null;
}

export interface ScoreJustification {
  score?: number;
  résumé?: string;
  détails?: string[];
  [key: string]: unknown;
}

export interface OfferDetail extends OfferListItem {
  normalized_description: string | null;
  education_level: string | null;
  deadline_at: string | null;
  offer_url: string | null;
  score_justification: ScoreJustification | null;
  updated_at: string;
}

export interface PaginatedOffers {
  items: OfferListItem[];
  total: number;
  page: number;
  page_size: number;
  has_next: boolean;
}

export interface SourceOut {
  id: string;
  name: string;
  source_type: string;
  base_url: string | null;
  is_active: boolean;
  check_frequency_hours: number;
}
